test/e2e/scheduling/taints.go

```diff
@@ -318,3 +324,126 @@ var _ = SIGDescribe("NoExecuteTaintManager [Serial]", func() {
+		if nodeName2 != nodeName1 {
+			framework.AddOrUpdateTaintOnNode(cs, nodeName2, testTaint)
+			framework.ExpectNodeHasTaint(cs, nodeName2, &testTaint)
+			defer framework.RemoveTaintOffNode(cs, nodeName2, testTaint)
+		}
+
+		// Wait a bit
+		By("Waiting for Pod1 to be deleted")
+		timeoutChannel := time.NewTimer(time.Duration(KubeletPodDeletionDelaySeconds+AdditionalWaitPerDeleteSeconds) * time.Second).C
+		var evicted int
+		for {
+			select {
+			case <-timeoutChannel:
+				if evicted == 0 {
+					framework.Failf("Failed to evict Pod1.")
+				} else if evicted == 2 {
+					framework.Failf("Pod1 is evicted. But unexpected Pod2 also get evicted.")
+				}
+				return
+			case podName := <-observedDeletions:
+				evicted++
+				if podName == podGroup+"1" {
+					framework.Logf("Noticed Pod %q gets evicted.", podName)
+				} else if podName == podGroup+"2" {
+					framework.Failf("Unexepected Pod %q gets evicted.", podName)
+					return
+				}
+			}
+		}
+	})
+
+	// 1. Run two pods both with toleration; one with tolerationSeconds=5, the other with 25
+	// 2. Taint the nodes running those pods with a no-execute taint
+	// 3. See if both pods get evicted in between [5, 25] seconds
+	It("evicts pods with minTolerationSeconds", func() {
+		podGroup := "taint-eviction-b"
+		observedDeletions := make(chan string, 100)
+		stopCh := make(chan struct{})
+		createTestController(cs, observedDeletions, stopCh, podGroup, ns)
+
+		pod1 := createPodForTaintsTest(true, AdditionalWaitPerDeleteSeconds, podGroup+"1", podGroup, ns)
+		pod2 := createPodForTaintsTest(true, 5*AdditionalWaitPerDeleteSeconds, podGroup+"2", podGroup, ns)
+
+		By("Starting pods...")
+		nodeName, err := testutils.RunPodAndGetNodeName(cs, pod1, 2*time.Minute)
+		framework.ExpectNoError(err)
+		framework.Logf("Pod1 is running on %v. Tainting Node", nodeName)
+		// ensure pod2 lands on the same node as pod1
+		pod2.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
+		_, err = testutils.RunPodAndGetNodeName(cs, pod2, 2*time.Minute)
+		framework.ExpectNoError(err)
+		framework.Logf("Pod2 is running on %v. Tainting Node", nodeName)
+
+		By("Trying to apply a taint on the Node")
+		testTaint := getTestTaint()
+		framework.AddOrUpdateTaintOnNode(cs, nodeName, testTaint)
+		framework.ExpectNodeHasTaint(cs, nodeName, &testTaint)
+		defer framework.RemoveTaintOffNode(cs, nodeName, testTaint)
+
+		// Wait a bit
+		By("Waiting for Pod1 and Pod2 to be deleted")
+		timeoutChannel := time.NewTimer(time.Duration(KubeletPodDeletionDelaySeconds+3*AdditionalWaitPerDeleteSeconds) * time.Second).C
+		var evicted int
+	FORLOOP:
+		for {
+			select {
+			case <-timeoutChannel:
+				framework.Failf("Failed to evict all Pods. %d pod(s) is not evicted.", 2-evicted)
+				return
+			case podName := <-observedDeletions:
+				framework.Logf("Noticed Pod %q gets evicted.", podName)
+				evicted++
+				if evicted == 2 {
+					break FORLOOP
+				}
+			}
+		}
+		if evicted < 2 {
```

Comment:
Given the above loop this will never be true.

Reply:
Exactly.